### PR TITLE
Heedls 515 remove delegate from group

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-515

### Description
Make the progress checkbox conditional and fix a layout issue with long group names.

### Screenshots
image
image

-----
### Developer checks
I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (controller, data services, services, view models etc) and manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [x] Updated/added documentation in Swiki and/or Readme.
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Data/DataServices/GroupsDataService.cs [hunk 1]
```diff
@@ -154,6 +156,24 @@ public void RemoveRelatedProgressRecordsForGroupDelegate(int groupId, int delega
...
+                          (SELECT ProgressID
+                            FROM Progress AS P
+                            INNER JOIN GroupCustomisations AS GC ON P.CustomisationID = GC.CustomisationID
+                            WHERE p.Completed IS NULL
+                                AND p.EnrollmentMethodID  = 3
+                                AND GC.GroupID = @groupId
+                                AND p.CandidateID = @delegateId
+                                AND P.RemovedDate IS NULL)",
+                new { groupId, delegateId }
+            ).FirstOrDefault();
```
Comment (stellake) [inline DigitalLearningSolutions.Data/DataServices/GroupsDataService.cs:174]: Could we just use the inner select? Also, would be good to have consistent capitalisation (usually lower case). :)

File: DigitalLearningSolutions.Data/DataServices/GroupsDataService.cs [hunk 2]
```diff
@@ -154,6 +156,24 @@ public void RemoveRelatedProgressRecordsForGroupDelegate(int groupId, int delega
...
+        {
+            return connection.Query<int?>(
+                @"SELECT ProgressId
+                    FROM Progress
+                    WHERE ProgressID IN
+                          (SELECT ProgressID
+                            FROM Progress AS P
+                            INNER JOIN GroupCustomisations AS GC ON P.CustomisationID = GC.CustomisationID
+                            WHERE p.Completed IS NULL
+                                AND p.EnrollmentMethodID  = 3
```
Comment (stellake) [inline DigitalLearningSolutions.Data/DataServices/GroupsDataService.cs:169]: Two spaces 👀

File: DigitalLearningSolutions.Data.Tests/DigitalLearningSolutions.Data.Tests.csproj [hunk 3]
```diff
@@ -21,6 +21,7 @@
     <PackageReference Include="FluentMigrator" Version="3.2.9" />
     <PackageReference Include="FluentMigrator.Runner" Version="3.2.9" />
     <PackageReference Include="FluentMigrator.Runner.SqlServer" Version="3.2.9" />
+    <PackageReference Include="JetBrains.DotMemoryUnit" Version="3.1.20200127.214830" />
```
Comment (stellake) [inline DigitalLearningSolutions.Data.Tests/DigitalLearningSolutions.Data.Tests.csproj:24]: Is this added on purpose/is it related to the ticket?

Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data.Tests/DigitalLearningSolutions.Data.Tests.csproj:24]: Nope, miss click when running a unit test, didn't realise it had added a package

**Review by stellake — APPROVED** [review]
Looks good!

**Pull request merged**